// @arcgis/map-components/dist/esm/arcgis-directions.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.0-next.37
 */
import { r as registerInstance, c as createEvent, h, g as getElement } from './index-6e172aac.js';
import { r as newLayersRouteLayer, s as newWidgetsDirections, i as importCoreReactiveUtils } from './index-0bc55ca5.js';
import { r as referenceElementHelper, i as isNullOrEmptyString, c as checkForView, a as addHandles } from './component-utils-23194f7c.js';
import { o as onDisconnectedHelper } from './disconnected-6cf1c100.js';

const directionsCss = "";

const loggerName = "map-components:arcgis-directions";
const ArcgisDirections = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisDirectionsReady = createEvent(this, "arcgisDirectionsReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this._watchHandles = [];
        this.referenceElement = undefined;
        this.apiKey = undefined;
        this.icon = undefined;
        this.routeLayerItemId = undefined;
        this.lastRoute = undefined;
        this.layer = undefined;
        this.maxStops = 50;
        this.searchProperties = undefined;
        this.unit = undefined;
        this.useDefaultRouteLayer = true;
        this.visibleElementsLayerDetails = true;
        this.visibleElementsSaveAsButton = true;
        this.visibleElementsSaveButton = true;
        this.position = "bottom-left";
        this.view = undefined;
        this.widget = undefined;
    }
    referenceElementWatcher(value) {
        void referenceElementHelper(this, value);
    }
    apiKeyWatcher(value) {
        if (value && this.widget) {
            this.widget.apiKey = value;
        }
    }
    iconWatcher(value) {
        if (value && this.widget) {
            this.widget.icon = value;
        }
    }
    routeLayerItemIdWatcher(value) {
        if (isNullOrEmptyString(value)) {
            this.view?.map.remove(this.layer);
            // maintain a default route layer
            if (this.useDefaultRouteLayer) {
                void newLayersRouteLayer({}).then((layer) => {
                    this.layer = layer;
                });
            }
        }
        else if (value && this.widget) {
            void newLayersRouteLayer({
                portalItem: {
                    id: value,
                },
            }).then((layer) => {
                this.layer = layer;
            });
        }
    }
    layerWatcher(value) {
        if (this.widget) {
            this.widget.layer = value ?? null;
        }
    }
    maxStopsWatcher(value) {
        if (value !== undefined && this.widget) {
            this.widget.maxStops = value;
        }
    }
    searchPropertiesWatcher(value) {
        if (this.widget) {
            this.widget.searchProperties = value;
        }
    }
    unitWatcher(value) {
        if (this.widget) {
            this.widget.unit = value;
        }
    }
    visibleElementsLayerDetailsWatcher(value) {
        if (this.widget) {
            this.widget.visibleElements = {
                ...this.widget.visibleElements,
                layerDetails: value,
            };
        }
    }
    visibleElementsSaveAsButtonWatcher(value) {
        if (this.widget) {
            this.widget.visibleElements = {
                ...this.widget.visibleElements,
                saveAsButton: value,
            };
        }
    }
    visibleElementsSaveButtonWatcher(value) {
        if (this.widget) {
            this.widget.visibleElements = {
                ...this.widget.visibleElements,
                saveButton: value,
            };
        }
    }
    positionWatcher(value) {
        if (value !== undefined && this.widget) {
            this.view.ui.move(this.childElem, value);
        }
    }
    viewWatcher(value) {
        if (value !== undefined && this.widget) {
            this.widget.view = value;
        }
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    async getDirections() {
        if (this.widget?.viewModel?.state !== "ready") {
            throw new Error(`${loggerName} - component not ready to get directions`);
        }
        return await this.widget?.getDirections();
    }
    async save() {
        if (this.widget?.viewModel?.state !== "ready") {
            throw new Error(`${loggerName} - component not ready to save directions`);
        }
        return await this.widget?.save();
    }
    async saveAs(portalItem, options) {
        if (this.widget?.viewModel?.state !== "ready") {
            throw new Error(`${loggerName} - component not ready to save directions as a Portal Item`);
        }
        return await this.widget?.saveAs(portalItem, options);
    }
    async zoomToRoute() {
        if (this.widget?.viewModel?.state !== "ready") {
            throw new Error(`${loggerName} - component not ready to zoom to route`);
        }
        this.widget?.zoomToRoute();
    }
    //--------------------------------------------------------------------------
    //
    //  Lifecycle
    //
    //--------------------------------------------------------------------------
    async componentDidLoad() {
        const params = await this._createParams();
        const widget = await newWidgetsDirections(params);
        this.widget = widget;
        await this._addWatchers();
        this._addContainer();
        this.arcgisDirectionsReady.emit();
        checkForView(this);
    }
    disconnectedCallback() {
        onDisconnectedHelper(this);
    }
    render() {
        return h("div", { ref: (el) => (this.childElem = el) });
    }
    //--------------------------------------------------------------------------
    //
    //  Private Methods
    //
    //--------------------------------------------------------------------------
    _addContainer() {
        this.widget.container = this.childElem;
    }
    async _addWatchers() {
        const { watch } = await importCoreReactiveUtils();
        addHandles(this, watch(() => this.widget.lastRoute, (value) => {
            this.lastRoute = value;
            this.arcgisPropertyChange.emit({ name: "lastRoute" });
        }, {
            initial: true,
        }));
    }
    async _createParams() {
        const params = {};
        params.visibleElements = {
            layerDetails: this.visibleElementsLayerDetails,
            saveAsButton: this.visibleElementsSaveAsButton,
            saveButton: this.visibleElementsSaveButton,
        };
        if (this.view) {
            params.view = this.view;
        }
        if (this.apiKey) {
            params.apiKey = this.apiKey;
        }
        if (this.icon) {
            params.icon = this.icon;
        }
        if (this.routeLayerItemId) {
            params.layer = await newLayersRouteLayer({
                portalItem: {
                    id: this.routeLayerItemId,
                },
            });
            this.layer = params.layer;
        }
        else if (this.layer) {
            params.layer = this.layer;
            this.layer = params.layer;
        }
        else if (this.useDefaultRouteLayer) {
            params.layer = await newLayersRouteLayer({});
            this.layer = params.layer;
        }
        if (this.maxStops !== undefined) {
            params.maxStops = this.maxStops;
        }
        if (this.searchProperties) {
            params.searchProperties = this.searchProperties;
        }
        if (this.unit) {
            params.unit = this.unit;
        }
        return params;
    }
    get el() { return getElement(this); }
    static get watchers() { return {
        "referenceElement": ["referenceElementWatcher"],
        "apiKey": ["apiKeyWatcher"],
        "icon": ["iconWatcher"],
        "routeLayerItemId": ["routeLayerItemIdWatcher"],
        "layer": ["layerWatcher"],
        "maxStops": ["maxStopsWatcher"],
        "searchProperties": ["searchPropertiesWatcher"],
        "unit": ["unitWatcher"],
        "visibleElementsLayerDetails": ["visibleElementsLayerDetailsWatcher"],
        "visibleElementsSaveAsButton": ["visibleElementsSaveAsButtonWatcher"],
        "visibleElementsSaveButton": ["visibleElementsSaveButtonWatcher"],
        "position": ["positionWatcher"],
        "view": ["viewWatcher"]
    }; }
};
ArcgisDirections.style = directionsCss;

export { ArcgisDirections as arcgis_directions };
