// @arcgis/map-components/dist/esm/arcgis-placement.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.0-next.37
 */
import { r as registerInstance, c as createEvent, h, g as getElement } from './index-6e172aac.js';
import { r as referenceElementHelper, c as checkForView } from './component-utils-23194f7c.js';
import { o as onDisconnectedHelper } from './disconnected-6cf1c100.js';

const placementCss = "arcgis-placement{}";

const ArcgisPlacement = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.placementReady = createEvent(this, "placementReady", 7);
        this.referenceElement = undefined;
        this.view = undefined;
        this.position = "bottom-left";
    }
    referenceElementWatcher(newValue) {
        void referenceElementHelper(this, newValue);
    }
    positionWatcher(newValue) {
        if (newValue !== undefined) {
            this.view.ui.move(this.childElem, newValue);
        }
    }
    //--------------------------------------------------------------------------
    //
    //  Lifecycle
    //
    //--------------------------------------------------------------------------
    async componentWillLoad() {
        this.placementReady.emit({ ready: true });
        checkForView(this);
    }
    disconnectedCallback() {
        onDisconnectedHelper(this);
    }
    render() {
        return (h("div", { ref: (el) => (this.childElem = el) }, h("slot", null)));
    }
    get el() { return getElement(this); }
    static get watchers() { return {
        "referenceElement": ["referenceElementWatcher"],
        "position": ["positionWatcher"]
    }; }
};
ArcgisPlacement.style = placementCss;

export { ArcgisPlacement as arcgis_placement };
