// @arcgis/map-components/dist/components/arcgis-version-management.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.30/esri/copyright.txt for details.
 * v4.30.0-next.37
 */
import { proxyCustomElement, HTMLElement, h, Host } from '@stencil/core/internal/client';
import { i as importCoreReactiveUtils, a6 as newWidgetsVersionManagementVersionManagementViewModel } from './index2.js';
import { u as useControllerManager, c as useT9n } from './chunk-WQTCUXBS.js';
import { d as defineCustomElement$5 } from './version-management-service-item.js';
import { d as defineCustomElement$4 } from './version-management-version-item.js';
import { d as defineCustomElement$3 } from './version-management-version-list.js';
import { d as defineCustomElement$2 } from './version-management-version-properties.js';
import "@esri/calcite-components/dist/components/calcite-flow.js";
import "@esri/calcite-components/dist/components/calcite-flow-item.js";
import "@esri/calcite-components/dist/components/calcite-panel.js";
import "@esri/calcite-components/dist/components/calcite-notice.js";

const versionManagementCss = ".div-content{background-color:var(--calcite-ui-foreground-1);padding:0.75rem}.flow-main{width:330px}.notice{width:-webkit-fill-available}.panel-services{max-height:400px}.panel-versions{max-height:400px}";

const __cmpMeta = [1,"arcgis-version-management",{"label":[1025],"position":[1025],"view":[1040],"messages":[1040],"state":[32]},undefined,{"position":["positionWatcher"]}];
const ArcgisVersionManagement$1 = /*@__PURE__*/ proxyCustomElement(class ArcgisVersionManagement extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
this.constructor.__registerControllers = ()=>__cmpMeta[2]??{};
        this.__attachShadow();
        this.manager = useControllerManager(this);
        this._watchHandles = [];
        this.label = undefined;
        this.position = "top-right";
        this.view = undefined;
        this.messages = this.manager.readonly(useT9n({ blocking: true }));
        this.state = undefined;
    }
    positionWatcher(newValue) {
        if (newValue !== undefined && this.viewModel) {
            this.view.ui.move(this.el, newValue);
        }
    }
    async componentDidLoad() {
        const { watch } = await importCoreReactiveUtils();
        this._watchHandles.push(watch(() => this.viewModel.state, (state) => {
            this.state = state;
        }));
    }
    async componentWillLoad() {
        await this._checkForView();
        this.viewModel = await newWidgetsVersionManagementVersionManagementViewModel({
            view: this.view,
        });
        this.label = this.label ? this.label : this.messages.label;
    }
    disconnectedCallback() {
        this._watchHandles.forEach((handle) => {
            handle.remove();
        });
        this._watchHandles = [];
    }
    render() {
        const { messages, label, viewModel, viewModel: { state, loadError }, } = this;
        const arrayServiceNames = Array.from(this.viewModel.serviceNameLookup, ([url, name]) => ({ url, name }));
        return (h(Host, null, h("calcite-flow", { ref: (node) => {
                this.flowElement = node;
            }, class: "flow-main" }, h("calcite-flow-item", { heading: label }, h("calcite-panel", { class: "panel-services", loading: state === "loading" }, arrayServiceNames.map((service) => (h("arcgis-version-management-service-item", { flowElement: this.flowElement, heading: label, serviceUrl: service.url, strings: messages, viewModel: viewModel, onVersionItemActionClickedEventWithServiceUrl: (e) => {
                if (this.flowElement) {
                    this._handleVersionItemActionClick(e, this.flowElement);
                }
            }, onVersionListActionClickedEvent: (e) => {
                if (this.flowElement) {
                    this._handleVersionListActionClick(e, this.flowElement);
                }
            } }))), state === "disabled" ? (h("calcite-notice", { class: "notice", closable: false, kind: "warning", open: true, scale: "s", slot: "footer" }, h("div", { slot: "message" }, this._getLoadError(loadError)))) : undefined)))));
    }
    //--------------------------------------------------------------------------
    //
    //  Private Methods
    //
    //--------------------------------------------------------------------------
    async _checkForView() {
        const viewRef = this.el.closest("arcgis-map");
        if (viewRef?.view) {
            await viewRef.view.map.load();
            this.view = viewRef.view;
            this.view.ui.add(this.el, this.position);
        }
    }
    _createVersionPropertiesFlow(serviceUrl, versionInfo) {
        const { flowElement, viewModel } = this;
        if (!flowElement) {
            return null;
        }
        const versionPropertiesFlow = document.createElement("arcgis-version-management-version-properties");
        versionPropertiesFlow.flowElement = flowElement;
        versionPropertiesFlow.serviceUrl = serviceUrl;
        versionPropertiesFlow.strings = this.messages;
        versionPropertiesFlow.versionInfo = versionInfo;
        versionPropertiesFlow.viewModel = viewModel;
        versionPropertiesFlow.addEventListener("versionPropertiesSaveClickedEvent", async (e) => {
            const { serviceUrl, versionInfo } = e.detail;
            const fullVersionName = versionInfo.versionIdentifier.name;
            const versionName = fullVersionName.split(".").length > 1 ? fullVersionName.split(".")[1] : fullVersionName;
            const ownerName = fullVersionName.split(".").length > 1 ? fullVersionName.split(".")[0] : viewModel.userLookup.get(serviceUrl);
            if (versionInfo.versionIdentifier.guid === "") {
                await viewModel.createVersion({
                    access: versionInfo.access,
                    description: versionInfo.description,
                    featureServerUrl: serviceUrl,
                    ownerName,
                    versionName,
                });
            }
            else {
                await viewModel.alterVersion({
                    access: versionInfo.access,
                    description: versionInfo.description,
                    featureServerUrl: serviceUrl,
                    ownerName,
                    versionIdentifier: {
                        name: "",
                        guid: versionInfo.versionIdentifier.guid,
                    },
                    versionName,
                });
            }
            if (viewModel.state === "ready") {
                const versionList = flowElement.getElementsByTagName("arcgis-version-management-version-list")[0];
                versionList.versionInfos = [];
                const versionInfos = await viewModel.getVersionInfos(serviceUrl, true);
                versionList.versionInfos = versionInfos;
                await flowElement.back();
            }
        });
        return versionPropertiesFlow;
    }
    _getLoadError(loadError) {
        const { messages } = this;
        switch (loadError) {
            case "no-feature-services":
                return messages.loadErrors.noFeatureServices;
            case "no-layers-property":
                return messages.loadErrors.noLayersProperty;
            default:
                return loadError;
        }
    }
    async _handleVersionItemActionClick(e, flowElement) {
        const { actionType, serviceUrl, versionInfo } = e.detail;
        switch (actionType) {
            case "changeVersion":
                await this._switchToVersion(serviceUrl, versionInfo.versionIdentifier.name, versionInfo.versionIdentifier.guid);
                break;
            case "editVersion": {
                const versionPropertiesFlow = this._createVersionPropertiesFlow(serviceUrl, versionInfo);
                if (versionPropertiesFlow) {
                    flowElement.append(versionPropertiesFlow);
                }
                break;
            }
        }
    }
    async _handleVersionListActionClick(e, flowElement) {
        const { actionType, serviceUrl } = e.detail;
        switch (actionType) {
            case "newVersion": {
                const versionPropertiesFlow = this._createVersionPropertiesFlow(serviceUrl, undefined);
                if (versionPropertiesFlow) {
                    flowElement.append(versionPropertiesFlow);
                }
                break;
            }
            case "refreshVersions": {
                if (this.flowElement) {
                    const versionList = this.flowElement.getElementsByTagName("arcgis-version-management-version-list")[0];
                    versionList.versionInfos = [];
                    const versionInfos = await this.viewModel.getVersionInfos(serviceUrl, true);
                    versionList.versionInfos = versionInfos;
                }
                break;
            }
        }
    }
    async _switchToVersion(featureServerUrl, toVersionName, toVersionGuid) {
        await this.viewModel.changeVersion(featureServerUrl, toVersionName, toVersionGuid);
    }
    static get assetsDirs() { return ["assets"]; }
    get el() { return this; }
    static get watchers() { return {
        "position": ["positionWatcher"]
    }; }
    static get style() { return versionManagementCss; }
}, __cmpMeta);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arcgis-version-management", "arcgis-version-management-service-item", "arcgis-version-management-version-item", "arcgis-version-management-version-list", "arcgis-version-management-version-properties"];
    components.forEach(tagName => { switch (tagName) {
        case "arcgis-version-management":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArcgisVersionManagement$1);
            }
            break;
        case "arcgis-version-management-service-item":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "arcgis-version-management-version-item":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "arcgis-version-management-version-list":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "arcgis-version-management-version-properties":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}
defineCustomElement$1();

const ArcgisVersionManagement = ArcgisVersionManagement$1;
const defineCustomElement = defineCustomElement$1;

export { ArcgisVersionManagement, defineCustomElement };
