// @esri/calcite-components/dist/esm/en-ca-3de639a5.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/main/LICENSE.md for details.
 * v2.8.5
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-e39bfd2b.js';
import './index-084e6474.js';
import './form-4b51b128.js';
import './dom-8c012086.js';
import './guid-71ff9371.js';
import './resources-91914bfd.js';
import './interactive-8e4cb61e.js';
import './browser-afd17c9c.js';
import './key-d339e1f0.js';
import './label-43c2e846.js';
import './component-320e2e16.js';
import './loadable-d76fe6f4.js';
import './locale-3bacb738.js';
import './observers-128a4167.js';
import './focusTrapComponent-6139276a.js';
import './math-400a5b13.js';
import './t9n-639aa12c.js';
import './openCloseComponent-a22be3a1.js';
import './Validation-18c00041.js';

// English (Canada) [en-ca]
var locale = {
  name: 'en-ca',
  weekdays: 'Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday'.split('_'),
  months: 'January_February_March_April_May_June_July_August_September_October_November_December'.split('_'),
  weekdaysShort: 'Sun_Mon_Tue_Wed_Thu_Fri_Sat'.split('_'),
  monthsShort: 'Jan_Feb_Mar_Apr_May_Jun_Jul_Aug_Sep_Oct_Nov_Dec'.split('_'),
  weekdaysMin: 'Su_Mo_Tu_We_Th_Fr_Sa'.split('_'),
  ordinal: function ordinal(n) {
    return n;
  },
  formats: {
    LT: 'h:mm A',
    LTS: 'h:mm:ss A',
    L: 'YYYY-MM-DD',
    LL: 'MMMM D, YYYY',
    LLL: 'MMMM D, YYYY h:mm A',
    LLLL: 'dddd, MMMM D, YYYY h:mm A'
  },
  relativeTime: {
    future: 'in %s',
    past: '%s ago',
    s: 'a few seconds',
    m: 'a minute',
    mm: '%d minutes',
    h: 'an hour',
    hh: '%d hours',
    d: 'a day',
    dd: '%d days',
    M: 'a month',
    MM: '%d months',
    y: 'a year',
    yy: '%d years'
  }
};
dayjs.locale(locale, null, true);

export { locale as default };
