// @esri/calcite-components/dist/esm/es-mx-2029da3d.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/main/LICENSE.md for details.
 * v2.8.5
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-e39bfd2b.js';
import './index-084e6474.js';
import './form-4b51b128.js';
import './dom-8c012086.js';
import './guid-71ff9371.js';
import './resources-91914bfd.js';
import './interactive-8e4cb61e.js';
import './browser-afd17c9c.js';
import './key-d339e1f0.js';
import './label-43c2e846.js';
import './component-320e2e16.js';
import './loadable-d76fe6f4.js';
import './locale-3bacb738.js';
import './observers-128a4167.js';
import './focusTrapComponent-6139276a.js';
import './math-400a5b13.js';
import './t9n-639aa12c.js';
import './openCloseComponent-a22be3a1.js';
import './Validation-18c00041.js';

// Spanish (Mexico) [es-mx]
var locale = {
  name: 'es-mx',
  weekdays: 'domingo_lunes_martes_miércoles_jueves_viernes_sábado'.split('_'),
  weekdaysShort: 'dom._lun._mar._mié._jue._vie._sáb.'.split('_'),
  weekdaysMin: 'do_lu_ma_mi_ju_vi_sá'.split('_'),
  months: 'enero_febrero_marzo_abril_mayo_junio_julio_agosto_septiembre_octubre_noviembre_diciembre'.split('_'),
  monthsShort: 'ene_feb_mar_abr_may_jun_jul_ago_sep_oct_nov_dic'.split('_'),
  relativeTime: {
    future: 'en %s',
    past: 'hace %s',
    s: 'unos segundos',
    m: 'un minuto',
    mm: '%d minutos',
    h: 'una hora',
    hh: '%d horas',
    d: 'un día',
    dd: '%d días',
    M: 'un mes',
    MM: '%d meses',
    y: 'un año',
    yy: '%d años'
  },
  ordinal: function ordinal(n) {
    return n + "\xBA";
  },
  formats: {
    LT: 'H:mm',
    LTS: 'H:mm:ss',
    L: 'DD/MM/YYYY',
    LL: 'D [de] MMMM [de] YYYY',
    LLL: 'D [de] MMMM [de] YYYY H:mm',
    LLLL: 'dddd, D [de] MMMM [de] YYYY H:mm'
  }
};
dayjs.locale(locale, null, true);

export { locale as default };
