// @esri/calcite-components/dist/esm/ja-3f275df3.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/main/LICENSE.md for details.
 * v2.8.5
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-e39bfd2b.js';
import './index-084e6474.js';
import './form-4b51b128.js';
import './dom-8c012086.js';
import './guid-71ff9371.js';
import './resources-91914bfd.js';
import './interactive-8e4cb61e.js';
import './browser-afd17c9c.js';
import './key-d339e1f0.js';
import './label-43c2e846.js';
import './component-320e2e16.js';
import './loadable-d76fe6f4.js';
import './locale-3bacb738.js';
import './observers-128a4167.js';
import './focusTrapComponent-6139276a.js';
import './math-400a5b13.js';
import './t9n-639aa12c.js';
import './openCloseComponent-a22be3a1.js';
import './Validation-18c00041.js';

// Japanese [ja]
var locale = {
  name: 'ja',
  weekdays: '日曜日_月曜日_火曜日_水曜日_木曜日_金曜日_土曜日'.split('_'),
  weekdaysShort: '日_月_火_水_木_金_土'.split('_'),
  weekdaysMin: '日_月_火_水_木_金_土'.split('_'),
  months: '1月_2月_3月_4月_5月_6月_7月_8月_9月_10月_11月_12月'.split('_'),
  monthsShort: '1月_2月_3月_4月_5月_6月_7月_8月_9月_10月_11月_12月'.split('_'),
  ordinal: function ordinal(n) {
    return n + "\u65E5";
  },
  formats: {
    LT: 'HH:mm',
    LTS: 'HH:mm:ss',
    L: 'YYYY/MM/DD',
    LL: 'YYYY年M月D日',
    LLL: 'YYYY年M月D日 HH:mm',
    LLLL: 'YYYY年M月D日 dddd HH:mm',
    l: 'YYYY/MM/DD',
    ll: 'YYYY年M月D日',
    lll: 'YYYY年M月D日 HH:mm',
    llll: 'YYYY年M月D日(ddd) HH:mm'
  },
  meridiem: function meridiem(hour) {
    return hour < 12 ? '午前' : '午後';
  },
  relativeTime: {
    future: '%s後',
    past: '%s前',
    s: '数秒',
    m: '1分',
    mm: '%d分',
    h: '1時間',
    hh: '%d時間',
    d: '1日',
    dd: '%d日',
    M: '1ヶ月',
    MM: '%dヶ月',
    y: '1年',
    yy: '%d年'
  }
};
dayjs.locale(locale, null, true);

export { locale as default };
